openedx/core/djangoapps/ace_common/settings/common.py

```diff
@@ -12,6 +12,6 @@ def plugin_settings(settings):
     settings.ACE_CHANNEL_DEFAULT_EMAIL = 'django_email'
     settings.ACE_CHANNEL_TRANSACTIONAL_EMAIL = 'django_email'
 
-    settings.ACE_ROUTING_KEY = 'edx.core.low'
+    settings.ACE_ROUTING_KEY = 'edx.cms.core.default'
```

Comment:
I'm not sure we should be hard coding this, versus getting the value from the config. I need a second opinion since I don't understand how our config works well enough.

Reply:
Actually I think @awaisdar001 asked for this line to be removed. I don't think this is the right queue to use for this anyways and we should be using the setting from the config used elsewhere for ACE. I think this line should already be doing that.

https://github.com/edx/edx-platform/blob/945debb00fef627f42772cd535e6213bf4db0bb5/openedx/core/djangoapps/ace_common/settings/aws.py#L16

Reply:
@jdmulloy I and @nadeemshahzad  Discussed and looked at this issue and decided the quick fix could be this b/c it is impacting all self paced courses. Regarding your question, I think I will let Nadeem answer it. 

@nadeemshahzad Can you please provide insight?

Reply:
I think we need to do something like the following:

1. Modify [this file](https://github.com/edx/edx-platform/blob/a439d5164c07e4695181b15244e8e5c7681421c4/openedx/core/djangoapps/ace_common/apps.py#L17) to look something like this:
```python
    plugin_app = {
        PluginSettings.CONFIG: {
            ProjectType.LMS: {
                SettingsType.AWS: {PluginSettings.RELATIVE_PATH: u'settings.aws'},
                SettingsType.COMMON: {PluginSettings.RELATIVE_PATH: u'settings.common'},
                SettingsType.DEVSTACK: {PluginSettings.RELATIVE_PATH: u'settings.devstack'},
            },
            ProjectType.CMS: {
                SettingsType.AWS: {PluginSettings.RELATIVE_PATH: u'settings.aws'},
                SettingsType.COMMON: {PluginSettings.RELATIVE_PATH: u'settings.common'},
                SettingsType.DEVSTACK: {PluginSettings.RELATIVE_PATH: u'settings.devstack'},
            },
        }
    }
```

My hypothesis is that none of these ACE specific configuration files are being loaded in the CMS because all of our non-plugin settings files do something [like this](https://github.com/edx/edx-platform/blob/master/cms/envs/aws.py#L593):

```python
plugin_settings.add_plugins(__name__, plugin_constants.ProjectType.CMS, plugin_constants.SettingsType.AWS)
```

And since the `ace_common` app doesn't define any settings for `ProjectType.CMS` it isn't getting any setting for ACE_ROUTING_KEY.

If we make the change above, I suspect CMS will start routing it's ACE tasks using the key `edx.lms.core.ace` defined [in edx-internal](https://github.com/edx/edx-internal/blob/master/ansible/vars/edx.yml#L673).

If that doesn't work, for some reason (I think it will) we can, instead, introduce a new setting that routes CMS tasks to a different queue.

Reply:
Actually, @cpennington raised the very good point that this task should not be using the ACE routing key. We should change this line:

https://github.com/edx/edx-platform/blob/master/openedx/core/djangoapps/schedules/tasks.py#L43

So that it uses the default CMS routing key in CMS and the default LMS routing key in LMS.

Or maybe if we remove the explicit routing key it will just use the default?

Reply:
Looking at the code more closely, I think we can just remove the routing_key argument from the task definition:

`@task(base=LoggedPersistOnFailureTask, bind=True, default_retry_delay=30)`

Reply:
@mulby / @jdmulloy I have updated the code according to your suggestion. Also spinned up a sandbox: https://tools-edx-jenkins.edx.org/view/All/job/Sandboxes/job/CreateSandbox/17304/ to test the fix.

Reply:
@mulby / @cpennington I have tested the fix on sandbox and it's working fine.

**sandbox links:**
`studio schedule and details page:` https://studio-edu-3965-update-course-schedule.sandbox.edx.org/settings/details/course-v1:edX+CN222+1T2019
`studio admin:` https://studio-edu-3965-update-course-schedule.sandbox.edx.org/admin/schedules/schedule/